sfge/src/main/java/com/salesforce/graph/ops/expander/PathExpansionRegistry.java

```diff
@@ -56,9 +61,34 @@ private static class ForkEventRegistryData extends RegistryData<ForkEvent> {
         // Nothing new to add
     }
 
-    /** Registry data structure to hold {@link ApexPathExpander}. */
+    /**
+     * Registry data structure to hold {@link ApexPathExpander}. See {@link
+     * com.salesforce.graph.ops.registry.RegistryDataLimitCalculator} for more information on how
+     * limit is set.
+     */
     private static class ApexPathExpanderRegistryData extends RegistryData<ApexPathExpander> {
-        // Nothing new to add
+        final int pathExpansionLimit;
+        final boolean isPathExpansionLimitSet;
+
+        ApexPathExpanderRegistryData() {
+            pathExpansionLimit = SfgeConfigProvider.get().getPathExpansionLimit();
+            isPathExpansionLimitSet = pathExpansionLimit > 0;
+        }
+
+        @Override
+        public void validateAndPut(ApexPathExpander instance) {
```

Comment:
We check the registry size each time before adding more items. If there's no limit set, we directly add the item.